common/src/main/java/com/wynntils/functions/generic/TimeFunctions.java

```diff
@@ -115,4 +116,29 @@ protected List<String> getAliases() {
             return List.of("offset");
         }
     }
+
+    public static class FormatTimeAdvancedFunction extends GenericFunction<String> {
+        @Override
+        public String getValue(FunctionArguments arguments) {
+            Time timestamp = arguments.getArgument("timestamp").getTime();
+            String format = arguments.getArgument("format").getStringValue();
+
+            try {
+                return new SimpleDateFormat(format).format(timestamp.timestamp());
+            } catch (IllegalArgumentException e) {
+                return "Invalid Format";
+            }
+        }
+
+        @Override
+        public FunctionArguments.RequiredArgumentBuilder getRequiredArgumentsBuilder() {
+            return new FunctionArguments.RequiredArgumentBuilder(List.of(
+                    new Argument<>("timestamp", Time.class, null), new Argument<>("format", String.class, null)));
```

Comment:
One more fix: please call Time variables `time`, and not `timestamp`. I've tried to reserve that for the actual long value that you can extract from a time.